pkg/local_object_storage/shard/exists.go

```diff
@@ -34,13 +36,26 @@ func (p *ExistsRes) Exists() bool {
 // Returns any error encountered that does not allow to
 // unambiguously determine the presence of an object.
 func (s *Shard) Exists(prm *ExistsPrm) (*ExistsRes, error) {
-	exists, err := s.objectExists(prm.addr)
+	exists, err := meta.Exists(s.metaBase, prm.addr)
+	if err != nil {
+		// If the shard is in degraded mode, try to consult blobstor directly.
+		// Otherwise, just return an error.
+		if s.GetMode() == ModeReadOnly {
```

Comment:
why do you try reading only in readOnly mode?

Reply:
That is the whole point of this PR: enable additional logic in degraded mode.